client/scripts/views/components/notification_row.ts

```diff
@@ -249,17 +269,30 @@ const NotificationRow: m.Component<{ notifications: Notification[] }, {
       }
       return m('li.NotificationRow', {
         class: notification.isRead ? '' : 'unread',
+        key: notification.id,
+        id: notification.id,
         onclick: async () => {
+          if (vnode.state.scrollOrStop) { vnode.state.scrollOrStop = false; return; }
           const notificationArray: Notification[] = [];
           notificationArray.push(notification);
           app.user.notifications.markAsRead(notificationArray).then(() => m.redraw());
-          if (!label.linkUrl) return;
-          await m.route.set(label.linkUrl);
+          await m.route.set(`/${app.activeId() || 'edgeware'}/notificationsList?id=${notification.id}`);
           m.redraw.sync();
         },
       }, [
         m('.comment-body', [
-          m('.comment-body-top', `${label.heading} on ${chainName}`),
+          m('.comment-body-top.chain-event-notification-top', [
+            `${label.heading} on ${chainName}`,
+            !vnode.attrs.onListPage && m(Icon, {
```

Comment:
Clicking X immediately closes the notification box and it's not clear whether the notification was deleted or not...would it be possible to update the notifications list without closing the parent popover?

Reply:
The parent "PopoverMenu" component had the property `closeOnContentClick` set to `true` which I set to `false`. The "Infinite" component is a bit finicky in regards to how it updates when the data its fed changes, ie it doesn't update unless its `key` changes alongside the data. What results is a small flash like on the NotificationsList page when clicking the X.

Reply:
I'm not sure this is a "problem" but it is noticeable.

Reply:
I'll put in a change to reduce the max number of notification rows shown, which should mitigate that flash significantly.